docs: list minimal permissions for Constellation setup (#1442)
edgelesssys/constellation: docs/docs/workflows/config.md

> You must be authenticated with the [Azure CLI](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli) in the shell session.
> 
> Your Azure account also needs to have at least the following permissions:

Reviewer: ```suggestion
You must be authenticated with the [Azure CLI](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli) in the shell session, and you need at least the following permissions:
```
nit: I'd shorten and link this to the previos sentence, but you can keep it if you prefer.

* `Microsoft.Authorization/roleDefinitions/*`
* `Microsoft.Authorization/roleAssignments/*`

> * `*/register/action`

Reviewer 2: I think this roughly needs to be in sync with the required subscription providers:
https://docs.edgeless.systems/constellation/getting-started/install#required-permissions

`*/register/action` can be reduced to the ones listed there under the condition that you run all Terraform commands with `ARM_SKIP_PROVIDER_REGISTRATION=true` (see the [troubleshooting](https://docs.edgeless.systems/constellation/workflows/troubleshooting#azure-resource-providers-cant-be-registered) link I recently added)

Author: > `*/register/action` can be omitted if you run all Terraform-related commands with `ARM_SKIP_PROVIDER_REGISTRATION=true`

Should I add this to the Azure text? Or would you completely replace it? And for "syncing" it with the resource providers listed at the installation step, I think the information there is incomplete? At least, if I try removing the `*/register/action` permission and run the command, I receive an error about at least 50 missing resource provider registrations from TF. (including the ones listed in your link though)

Reviewer 2: Yes, this is why you need to set `ARM_SKIP_PROVIDER_REGISTRATION=true` in this case. 

The Azure Terraform provider has this slightly dumb behavior that it tries to register all resource providers it provides even if it never uses them. This makes it require way more permissions _by default_. But if you set `ARM_SKIP_PROVIDER_REGISTRATION=true` and register the actually required providers manually, the list of requires permissions is way lower (and if you manually register, you don't need `*/register/action`.

So given that this requires manual intervention, we could maybe put a disclaimer here saying that `*/register/action` is not needed if you set `ARM_SKIP_PROVIDER_REGISTRATION=true`, but you need to manually register the resource providers.
